osu.Game/Screens/OnlinePlay/Multiplayer/MultiplayerMatchSubScreen.cs

```diff
@@ -321,6 +323,26 @@ private void onUserModsChanged(ValueChangedEvent<IReadOnlyList<Mod>> mods)
             client.ChangeUserMods(mods.NewValue);
         }
 
+        private void updateBeatmapAvailability(ValueChangedEvent<BeatmapAvailability> _ = null)
+        {
+            if (client.Room == null)
+                return;
+
+            client.ChangeBeatmapAvailability(BeatmapAvailability.Value);
+
+            if (client.LocalUser?.State == MultiplayerUserState.Ready)
+                client.ChangeState(MultiplayerUserState.Idle);
+        }
+
+        private void onRoomUpdated()
+        {
+            if (client.Room == null)
+                return;
+
+            if (client.LocalUser?.BeatmapAvailability.Equals(BeatmapAvailability.Value) == false)
+                updateBeatmapAvailability();
+        }
```

Comment:
i'm confused. what is the null argument on this method and stuff i don't know what's going on here.

Reply:
basically `updateBeatmapAvailability` may be called while `client.Room = null`, so I need to guard against that, and subscribe to `RoomUpdated` for updating current availability to the joined room.

Reply:
roomupdated is called when users join and leave the room and stuff. is this just a catch all to handle beatmap changes...?

you really have to describe this kihd of stuff in yyour code using comments because it reads horribly.

Reply:
wait no. that case should be handled by the availablility tracker? i'm lost.

Reply:
is it safe to remove this?

Reply:
hmm, you mean because when a room is joined the playlist is refreshed and selected item will therefore update? yeah then.

Reply:
I'm thoroughly confused.

Reply:
Removed for now, seems to still work correctly.